kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/filters/AuthorizationST.java

```diff
@@ -0,0 +1,229 @@
+import io.kroxylicious.systemtests.Constants;
+import io.kroxylicious.systemtests.Environment;
+import io.kroxylicious.systemtests.clients.records.ConsumerRecord;
+import io.kroxylicious.systemtests.enums.KafkaClientType;
+import io.kroxylicious.systemtests.installation.kroxylicious.Kroxylicious;
+import io.kroxylicious.systemtests.installation.kroxylicious.KroxyliciousOperator;
+import io.kroxylicious.systemtests.steps.KafkaSteps;
+import io.kroxylicious.systemtests.steps.KroxyliciousSteps;
+import io.kroxylicious.systemtests.templates.strimzi.KafkaNodePoolTemplates;
+import io.kroxylicious.systemtests.templates.strimzi.KafkaTemplates;
+import io.kroxylicious.systemtests.utils.KafkaUtils;
+
+import static io.kroxylicious.systemtests.k8s.KubeClusterResource.kubeClient;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assumptions.assumeThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+class AuthorizationST extends AbstractST {
+    protected static final String BROKER_NODE_NAME = "kafka";
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuthorizationST.class);
+    private static final String MESSAGE = "Hello-world";
+    private final String clusterName = "my-cluster";
+    private String bootstrap;
+    private KroxyliciousOperator kroxyliciousOperator;
+    private static Kroxylicious kroxylicious;
+    private final Map<String, String> usernamePasswords = new HashMap<>();
+    List<String> aclRules;
+
+    @BeforeAll
+    void setUp() {
+        List<Pod> kafkaPods = kubeClient().listPodsByPrefixInName(Constants.KAFKA_DEFAULT_NAMESPACE, clusterName);
+        if (!kafkaPods.isEmpty()) {
+            LOGGER.atInfo().setMessage("Skipping kafka deployment. It is already deployed!").log();
+        }
+        else {
+            LOGGER.atInfo().setMessage("Deploying Kafka in {} namespace").addArgument(Constants.KAFKA_DEFAULT_NAMESPACE).log();
+
+            int numberOfBrokers = 1;
+            KafkaBuilder kafka = KafkaTemplates.kafkaPersistentWithAuthentication(Constants.KAFKA_DEFAULT_NAMESPACE, clusterName, numberOfBrokers);
+
+            resourceManager.createResourceFromBuilderWithWait(
+                    KafkaNodePoolTemplates.kafkaBasedNodePoolWithDualRole(BROKER_NODE_NAME, kafka.build(), numberOfBrokers),
+                    kafka);
+        }
+
+        kroxyliciousOperator = new KroxyliciousOperator(Constants.KROXYLICIOUS_OPERATOR_NAMESPACE);
+        kroxyliciousOperator.deploy();
+
+        generatePasswordForNewUser(Constants.KROXYLICIOUS_ADMIN_USER);
+    }
+
+    @BeforeEach
+    void beforeEach() {
+        bootstrap = null;
+        aclRules = new ArrayList<>();
+        aclRules.add(generateAllowAclRule(Constants.KROXYLICIOUS_ADMIN_USER, "*"));
+    }
+
+    @AfterAll
+    void cleanUp() {
+        kroxyliciousOperator.delete();
+    }
+
+    @Test
+    void testScramSha512AuthenticationAllowProduceAndConsume(String namespace) {
```

Comment:
Having `ScramSha512` in the test name feels a bit funky as its not mentioned again in the test body and is not configurable anywhere.